### ISSUE TITLE
Build the github actions to validate pull requests

### ISSUE THREAD

**Issue by Leofariasrj25** [NGINXcellent/webserv#1]
One of the building blocks of our development process is to validate a pull requests, reducing the cognitive load of checking for errors and giving us the ability to focus more on style and better code practices.

This also has the added benefit of allowing us gain more confidence as the time passes by and ensuring trust that the code we commit is actually good.

**Comment by dvargas1** [comment]
I have created a simple makefile and a main.cpp file to be placed at the root. Until we upload the final version, whenever a pull request is made to the main branch, it will go through the "make test" rule, and if it returns failure, the PR is automatically marked as failed.

I will continue researching more code checks to implement.

Sorry for not using the specific issue branch, but I needed to test directly on the main branch.
